lib/modules/asset/AssetsController.ts

```diff
@@ -227,20 +234,34 @@ export class AssetsController {
       request.context.connection.protocol === "http" &&
       request.context.connection.misc.verb === "GET"
     ) {
-      const exportId = request.getString("exportId");
-
-      const { id } = await this.measureExporter.getExport(engineId, exportId);
-
-      request.response.configure({
-        headers: {
-          "Content-Disposition": `attachment; filename="asset-${id}.csv"`,
-          "Content-Type": "text/csv",
-        },
-      });
-
-      const stream = await this.measureExporter.sendExport(engineId, exportId);
-
-      return new HttpStream(stream);
+      try {
+        const exportId = request.getString("exportId");
+
+        const { id } = await this.measureExporter.getExport(engineId, exportId);
+        const stream = await this.measureExporter.sendExport(
+          engineId,
+          exportId
+        );
+
+        request.response.configure({
+          headers: {
+            "Content-Disposition": `attachment; filename="asset-${id}.csv"`,
+            "Content-Type": "text/csv",
+          },
+        });
+
+        return new HttpStream(stream);
+      } catch (error) {
+        request.response.configure({
```

Comment:
Why adding this instead of throwing a traditional error ?

Reply:
Because this endpoint is mostly called by browser and traditional error return a JSON that can be hard to understand for user.
Maybe the better solution could be to improve global error handler in the core

Reply:
Add a comment to explain this in the code :)